lib/spack/spack/util/environment.py

```diff
@@ -968,7 +973,7 @@ def _source_single_file(file_and_args, environment):
         # go with sys.executable. Below we just need a working
         # Python interpreter, not necessarily sys.executable.
         python_cmd = executable.which('python3', 'python', 'python2')
-        python_cmd = python_cmd.name if python_cmd else sys.executable
+        python_cmd = python_cmd.path if python_cmd else sys.executable
```

Comment:
This is not strictly necessary after 9ace7f1f3ba24833dcc5dc0d7e91daa0941a3161, but below we're sourcing a file, which may change PATH, so it's better to work with absolute paths than names...